src/nupic/algorithms/KNNClassifier.py

```diff
@@ -985,11 +1013,23 @@ def _getDistances(self, inputPattern, partitionId=None):
 
 
   def finishLearning(self):
+    """
+    TODO: Somebody help me write this.
+
+    :return:
+    """
     if self.numSVDDims is not None and self._vt is None:
       self.computeSVD()
 
 
   def computeSVD(self, numSVDSamples=None, finalize=True):
+    """
+    TODO: Somebody help me write this.
+
+    :param numSVDSamples:
+    :param finalize:
+    :return:
+    """
```

Comment:
```

def computeSVD(self, numSVDSamples=None, finalize=True):
    """
    Compute the singular value decomposition (SVD). The SVD is a factorization of a real or complex matrix. It factors the matrix `a` as `u * np.diag(s) * v`, where `u` and `v` are unitary and `s` is a 1-d array of `a`'s singular values.

    Reason for computing the SVD: 
    There are cases where you want to feed a lot of vectors to the KNNClassifier. However, this can be slow. You can speed up training by (1) computing the SVD of the input patterns which will give you the eigenvectors, (2) only keeping a fraction of the eigenvectors, and (3) projecting the input patterns onto the remaining eigenvectors. 

    Note that all input patterns are projected onto the eigenvectors in the same fashion. Keeping only the highest eigenvectors increases training performance since it reduces the dimensionality of the input. 

    :param numSVDSamples: (int) the number of samples to use for the SVD computation.
    :param finalize: (bool) whether to apply SVD to the input patterns.
    :return s: (array) The singular values for every matrix, sorted in descending order.
    """
```